evap/results/tests/test_views.py

```diff
@@ -732,216 +749,126 @@ def test_private_evaluation(self):
-        self.assertNotIn(".responsible_contributor_orig_hidden.", page)
-        self.assertNotIn(".responsible_contributor_orig_published_changed.", page)
-        self.assertIn(".responsible_contributor_changed_published.", page)
-        self.assertIn(".responsible_contributor_orig_private.", page)
-        self.assertNotIn(".responsible_contributor_orig_notreviewed.", page)
-        self.assertIn(".responsible_contributor_additional_orig_published.", page)
-        self.assertNotIn(".responsible_contributor_additional_orig_hidden.", page)
-
-    def test_textanswer_visibility_for_manager(self):
-        page = self.app.get("/results/semester/1/evaluation/1?view=full", user=self.manager)
-        self.assertIn(".general_orig_published.", page)
-        self.assertNotIn(".general_orig_hidden.", page)
-        self.assertNotIn(".general_orig_published_changed.", page)
-        self.assertIn(".general_additional_orig_published.", page)
-        self.assertNotIn(".general_additional_orig_hidden.", page)
-        self.assertIn(".general_changed_published.", page)
-        self.assertIn(".contributor_orig_published.", page)
-        self.assertIn(".contributor_orig_private.", page)
-        self.assertIn(".responsible_contributor_orig_published.", page)
-        self.assertNotIn(".responsible_contributor_orig_hidden.", page)
-        self.assertNotIn(".responsible_contributor_orig_published_changed.", page)
-        self.assertIn(".responsible_contributor_changed_published.", page)
-        self.assertIn(".responsible_contributor_orig_private.", page)
-        self.assertNotIn(".responsible_contributor_orig_notreviewed.", page)
-        self.assertIn(".responsible_contributor_additional_orig_published.", page)
-        self.assertNotIn(".responsible_contributor_additional_orig_hidden.", page)
-
+    def check_with_view(
+        self,
+        user,
+        expected_visible_textanswers,
+        general=ViewGeneralResults,
+        contributor=ViewContributorResults,
+        all_textanswers=general_textanswers + contributor_textanswers,
+    ):
+        textanswers_not_in = list(set(all_textanswers) - set(expected_visible_textanswers))
+        for general_view, contributor_view in product(general, contributor):
+            page = self.app.get(
+                f"/results/semester/1/evaluation/1?view_general_results={general_view.value}&view_contributor_results={contributor_view.value}",
+                user=user,
+            )
 
-class TestResultsTextanswerVisibility(WebTest):
-    fixtures = ["minimal_test_data_results"]
+            for answer in expected_visible_textanswers:
+                self.assertIn(answer, page)
+            for answer in textanswers_not_in:
+                self.assertNotIn(answer, page)
+
+    def test_manager(self):
+        user = self.manager
+        self.check_with_view(user, [])
+        with run_in_staff_mode(self):  # in staff mode, the manager can see everything except deleted or changed
```

Comment:
comment doesn't match assertion below, we assert that everything from ` self.standard_general_textanswers` is visible, which includes `self.standard_general_textanswers`.

I would, without checking more deeply, expect that noone can ever see deleted or the originals of changed answers? In that case, I would also try to generalize that out, so that the logic here doesn't have to list any answers at all, but literally just says: "As a manager, I can see all comments that could possibly be shown to any user".